lib/ruby_lsp/document.rb

```diff
@@ -180,6 +181,18 @@ def sorbet_sigil_is_true_or_higher
       end
     end
 
+    sig { params(char_position: Integer).returns(Integer) }
+    def convert_to_byte_offset(char_position)
+      case encoding
+      when Encoding::UTF_8
+        source.slice(0, char_position).bytesize
+      when Encoding::UTF_16, Encoding::UTF_16LE, Encoding::UTF_16BE
+        char_position * 2
```

Comment:
Do we need to consider surrogate pairs?

Reply:
Thank you for answering. I will work on making the logic common and adding tests.
I'm not very familiar with the directory structure and rules of ruby-lsp, but is it okay if I add the implementation to `lib/ruby_lsp/utils.rb`?